### ISSUE TITLE
[Occasional connection failure] error: could not download release latest for this platform.

### ISSUE THREAD

**Issue by ShuhuaGao** [mitsuhiko/rye#561]
### Discussed in https://github.com/mitsuhiko/rye/discussions/560

<div type='discussions-op-text'>

<sup>Originally posted by **ShuhuaGao** January 26, 2024</sup>
```powershell
PS C:\Users\shuhu> rye self update
Updating to latest
error: could not download release latest for this platform

Caused by:
    0: download of https://github.com/mitsuhiko/rye/releases/latest/download/rye-x86_64-windows.exe failed
    1: [28] Timeout was reached (Failed to connect to github.com port 443 after 21045 ms: Couldn't connect to server)
```

However, I can access the link `https://github.com/mitsuhiko/rye/releases/latest/download/rye-x86_64-windows.exe` directly and download the .exe file with a browser. 

Before updating, the version info is
```powershell
PS C:\Users\shuhu> rye --version
rye 0.15.2
commit: 0.15.2 (e2beb3a90 2023-10-04)
platform: windows (x86_64)
self-python: cpython@3.11
symlink support: true
```

Extra information: I am in China and use VPN to access GitHub. The following configuration is used:
```powershell
PS C:\Users\shuhu> git config --global https.proxy
https://127.0.0.1:4780
PS C:\Users\shuhu> git config --global http.proxy
http://127.0.0.1:4780
```

**Comment by ShuhuaGao** [comment]
Further trial: I set the proxy in rye config and the result is 
```powershell
PS C:\Users\shuhu> rye config --get proxy.https
https://127.0.0.1:4780
PS C:\Users\shuhu> rye config --get proxy.http
http://127.0.0.1:4780
PS C:\Users\shuhu> rye self update
Updating to latest
error: could not download release latest for this platform

Caused by:
    0: download of https://github.com/mitsuhiko/rye/releases/latest/download/rye-x86_64-windows.exe failed
    1: [35] SSL connect error (schannel: failed to receive handshake, SSL/TLS connection failed)
```

**Comment by mitsuhiko** [comment]
The SSL failure is likely resolved by #536. Unfortunately you might need to re-install to benefit of this change. I am closing this as resolved given that the MITM issue has been fixed.
